pkg/executor/infoschema_reader.go

```diff
@@ -3448,6 +3454,203 @@ func (e *memtableRetriever) setDataFromKeywords() error {
+	for _, sv := range sysVars {
+		if infoschema.SysVarHiddenForSem(ctx, sv.Name) {
+			continue
+		}
+		currentVal, err := ctx.GetSessionVars().GetSessionOrGlobalSystemVar(context.Background(), sv.Name)
+		if err != nil {
+			currentVal = ""
+		}
+		instance := "*"
+		if sv.HasInstanceScope() {
+			serverInfo, err := infosync.GetServerInfo()
+			if err != nil {
+				logutil.BgLogger().Error("failed to get server info", zap.Error(err))
+				instance = "unknown"
+			} else {
+				instance = serverInfo.IP
+			}
+		}
+		isClusterDynamic := false
+		if sv.HasGlobalScope() {
+			isClusterDynamic = true
+		}
+		isInstDynamic := false
+		if sv.HasInstanceScope() {
+			isInstDynamic = true
+		}
+		isSessionDynamic := false
+		if sv.HasSessionScope() {
+			isSessionDynamic = true
+		}
+		row := types.MakeDatums(
+			"tidb",
+			"VAR",
+			sv.Name,
+			instance,
+			sv.Scope.String(),
+			currentVal,
+			sv.Value,
+			sv.MinValue,
+			sv.MaxValue,
+			nil,
+			isClusterDynamic,
+			isInstDynamic,
+			isSessionDynamic,
+		)
+		// min and max value is only supported for numeric types
+		if !(sv.Type == variable.TypeUnsigned || sv.Type == variable.TypeInt || sv.Type == variable.TypeFloat) {
+			row[7].SetNull()
+			row[8].SetNull()
+		}
+		if sv.Type == variable.TypeEnum {
+			possibleValues := strings.Join(sv.PossibleValues, ",")
+			row[9].SetString(possibleValues, mysql.DefaultCollationName)
+		}
+		rows = append(rows, row)
+	}
+
+	// Get configuration info
+	type result struct {
+		idx  int
+		rows [][]types.Datum
+		err  error
+	}
+	serversInfo, err := infoschema.GetClusterServerInfo(ctx)
+	failpoint.Inject("mockParamsServerInfo", func(val failpoint.Value) {
+		if s := val.(string); len(s) > 0 {
+			// erase the error
+			serversInfo, err = parseFailpointServerInfo(s), nil
+		}
+	})
+	if err != nil {
+		ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+		return nil
+	}
+	wg := sync.WaitGroup{}
+	ch := make(chan result, len(serversInfo))
+	for i, srv := range serversInfo {
+		typ := srv.ServerType
+		address := srv.Address
+		statusAddr := srv.StatusAddr
+		if len(statusAddr) == 0 {
+			ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("%s node %s does not contain status address", srv.ServerType, address))
+			continue
+		}
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+			var url string
+			switch typ {
+			case "pd":
+				url = fmt.Sprintf("%s://%s/pd/api/v1/config/detail", util.InternalHTTPSchema(), statusAddr)
+			case "tikv", "tidb", "tiflash":
+				url = fmt.Sprintf("%s://%s/config/detail", util.InternalHTTPSchema(), statusAddr)
+			default:
+				ch <- result{err: errors.Errorf("currently we do not support get config from node type: %s(%s)", typ, address)}
+				return
+			}
+			req, err := http.NewRequest(http.MethodGet, url, nil)
+			if err != nil {
+				ch <- result{err: errors.Trace(err)}
+				return
+			}
+			req.Header.Add("PD-Allow-follower-handle", "true")
+			resp, err := util.InternalHTTPClient().Do(req)
```

Comment:
Do we set a timeout on the HTTP client anywhere? and/or use a context with a timeout?

Reply:
If we set a timeout for it, that means we should offer a configuration item that allows the user to set the timeout time. This timeout should also be set for other places in tidb where `InternalHTTPClient ` is used